core/chains/evm/txmgr/orm.go

```diff
@@ -332,3 +341,137 @@ func (o *orm) UpdateBroadcastAts(now time.Time, etxIDs []int64) error {
 	_, err := o.q.Exec(`UPDATE eth_txes SET broadcast_at = $1 WHERE id = ANY($2) AND broadcast_at < $1`, now, pq.Array(etxIDs))
 	return errors.Wrap(err, "updateBroadcastAts failed to update eth_txes")
 }
+
+// SetBroadcastBeforeBlockNum updates already broadcast attempts with the
+// current block number. This is safe no matter how old the head is because if
+// the attempt is already broadcast it _must_ have been before this head.
+func (o *orm) SetBroadcastBeforeBlockNum(blockNum int64, chainID big.Int) error {
+	_, err := o.q.Exec(
+		`UPDATE eth_tx_attempts
+SET broadcast_before_block_num = $1 
+FROM eth_txes
+WHERE eth_tx_attempts.broadcast_before_block_num IS NULL AND eth_tx_attempts.state = 'broadcast'
+AND eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_txes.evm_chain_id = $2`,
+		blockNum, chainID.String(),
+	)
+	return errors.Wrap(err, "SetBroadcastBeforeBlockNum failed")
+}
+
+func (o *orm) FindEtxAttemptsConfirmedMissingReceipt(chainID big.Int) (attempts []EthTxAttempt, err error) {
+	err = o.q.Select(&attempts,
+		`SELECT DISTINCT ON (eth_tx_attempts.eth_tx_id) eth_tx_attempts.*
+		FROM eth_tx_attempts
+		JOIN eth_txes ON eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_txes.state = 'confirmed_missing_receipt'
+		WHERE evm_chain_id = $1
+		ORDER BY eth_tx_attempts.eth_tx_id ASC, eth_tx_attempts.gas_price DESC, eth_tx_attempts.gas_tip_cap DESC`,
+		chainID.String())
+	if err != nil {
+		err = errors.Wrap(err, "FindEtxAttemptsConfirmedMissingReceipt failed to query")
+	}
+	return
+}
+
+func (o *orm) UpdateEthTxsUnconfirmed(ids []int64) error {
+	_, err := o.q.Exec(`UPDATE eth_txes SET state='unconfirmed' WHERE id = ANY($1)`, pq.Array(ids))
+
+	if err != nil {
+		return errors.Wrap(err, "UpdateEthTxsUnconfirmed failed to execute")
+	}
+	return nil
+}
+
+func (o *orm) FindEthTxAttemptsRequiringReceiptFetch(chainID big.Int) (attempts []EthTxAttempt, err error) {
+	err = o.q.Transaction(func(tx pg.Queryer) error {
+		err = tx.Select(&attempts, `
+SELECT eth_tx_attempts.* FROM eth_tx_attempts
+JOIN eth_txes ON eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_txes.state IN ('unconfirmed', 'confirmed_missing_receipt') AND eth_txes.evm_chain_id = $1
+WHERE eth_tx_attempts.state != 'insufficient_eth'
+ORDER BY eth_txes.nonce ASC, eth_tx_attempts.gas_price DESC, eth_tx_attempts.gas_tip_cap DESC
+`, chainID.String())
+		if err != nil {
+			return errors.Wrap(err, "FindEthTxAttemptsRequiringReceiptFetch failed to load eth_tx_attempts")
+		}
+		err = loadEthTxes(tx, attempts)
```

Comment:
We use `loadEthTxes()` here which is a package-level private method in the [eth_confirmer.go](https://github.com/smartcontractkit/chainlink/blob/augustus.BCI-268.eth-confirmer-refactor/core/chains/evm/txmgr/eth_confirmer.go#L854) . However, there's also a private method in orm.go [here](https://github.com/smartcontractkit/chainlink/blob/augustus.BCI-268.eth-confirmer-refactor/core/chains/evm/txmgr/orm.go#L87) `preloadTxes` .

I was considering replacing loadEthTxes with preloadTxes, however elsewhere in the eth_confirmer.go we also do a call to loadEthTxes all by itself (with no corresponding adjacent SQL query) so I would need to either keep the current definition of loadEthTxes or create a public method on the orm struct. Any suggestions on what to do?

Reply:
Does `preloadTxes` do the same thing as `loadEthTxes`? If so I would consolidate and make it a public method on the ORM

Reply:
Yeah, it does exactly the same thing!

Reply:
Sounds good, I'll refactor to make it public

Reply:
I'll do so in a follow up PR. I think there are too many changes in this PR as is